src/os/bluestore/BlueStore.cc

```diff
@@ -1990,6 +1990,8 @@ void BlueStore::ExtentMap::dup(BlueStore* b, TransContext* txc,
   uint64_t& length, uint64_t& dstoff) {
 
   auto cct = onode->c->store->cct;
+  bool inject_21040 =
+    cct->_conf->get_val<bool>("bluestore_debug_inject_bug21040");  
```

Comment:
Can `ExtentMap::dup` appear on any hot path? If so, we might need caching as `md_config_t::get_val<T>` is pretty expensive.